src/aap_eda/core/models/base.py

```diff
@@ -0,0 +1,28 @@
+#  Copyright 2024 Red Hat, Inc.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+from django.db import models
+
+__all__ = ("BaseModel",)
+
+
+class BaseModel(models.Model):
+    class Meta:
+        abstract = True
+
+    def summary_fields(self):
+        return {
+            "id": self.id,
+            "name": self.name,
```

Comment:
It looks like it would be _viable_ to also consolidate

```python
name = models.TextField(null=False, unique=True)
```

Meaning, delete these fields from all subclasses of `BaseModel`. That is roughly approaching how AWX & DAB and other things did this. Instead of just a base model, it becomes a NamedModel, and that might start becoming familiar if you've seen the inheritance structure in other projects...

Doing this, you don't have to ask yourself the question "does this model have a name field".

I realize this won't work for some models now because they are exceptions. It would be good to actually dive into what drove those. Like, does `RulebookProcess` not have _any_ unique constraints? Is that really intended? For those models, maybe you should define `summary_fields` in the model instead of inheriting from the base.

Reply:
@AlanCoding Updated the PR. I left out the models that have `name` field defined differently. `Rulebook`, etc. have `unique_together` constraints for cases like two projects could have a rulebook with the same name, so we have to do `unique_together = ["project_id", "name"]`. As for `RulebookProcess`, it inherits name from its parent `Activation` so all RulebookProcess objects relating to one Activation all have the same name.